src/app/services/pagination.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+    providedIn: 'root'
+})
```

Comment:
I dont like this 'providedIn: 'root'', this param gives you possibility to use this service in a whole app.
In my opinion, you can use @Injectable() and add to the array of providers wheere you use this service.

Reply:
Done